include/boost/url/grammar/is_mutable_string.hpp

```diff
@@ -0,0 +1,78 @@
+//
+// Copyright (c) 2016-2019 Vinnie Falco (vinnie dot falco at gmail dot com)
+//
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+//
+// Official repository: https://github.com/CPPAlliance/url
+//
+
+#ifndef BOOST_URL_IS_MUTABLE_STRING_HPP
+#define BOOST_URL_IS_MUTABLE_STRING_HPP
+
+#include <type_traits>
+
+namespace boost {
+namespace urls {
+namespace grammar {
+
+/** Alias for `std::true_type` if `T` satisfies __MutableString__
+    for the iterator type `I`
+*/
+#ifdef BOOST_URL_DOCS
+template<class T, class I>
+using is_mutable_string = __see_below__;
+#else
+namespace detail
+{
+template<class T, class = void>
+struct has_value_type : std::false_type {};
```

Comment:
I think you could have done this all in `is_mutable_string`

Reply:
I tried to do that but I couldn't because of some other implementation details.